app/models/ont/library_factory.rb

```diff
@@ -0,0 +1,152 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/ClassLength
+
+# Ont namespace
+module Ont
+  # LibraryFactory
+  # The factory will create libraries from a given plate
+  class LibraryFactory
+    include ActiveModel::Model
+
+    validate :check_validation_errors, :check_libraries, :check_tag_taggables
+
+    def initialize(attributes = {})
+      @validation_errors = []
+      @libraries = []
+      @tag_taggables = []
+      @container_materials = []
+      build_libraries if fetch_and_validate_entities(attributes)
+    end
+
+    def save
+      return false unless valid?
+
+      @libraries.collect(&:save)
+      @tag_taggables.collect(&:save)
+      @container_materials.collect(&:save)
+      true
+    end
+
+    private
+
+    attr_reader :validation_errors,
+                :tag_set,
+                :plate,
+                :sorted_wells,
+                :tag_taggables,
+                :container_materials
+
+    def fetch_and_validate_entities(attributes)
+      try_get_tag_set(attributes[:tag_set_name]) &&
+        try_get_plate(attributes[:plate_barcode]) &&
+        validate_tag_set_for_plate &&
+        try_get_sorted_wells(attributes[:well_primary_grouping_direction])
+    end
+
+    def try_get_tag_set(tag_set_name)
+      @tag_set = ::TagSet.find_by(name: tag_set_name)
+      return true unless @tag_set.nil?
+
+      validation_errors << ['Tag set', 'must exist']
+      false
+    end
+
+    def try_get_plate(plate_barcode)
+      @plate = ::Plate.find_by(barcode: plate_barcode)
+      return true unless @plate.nil?
+
+      validation_errors << ['Plate', 'must exist']
+      false
+    end
+
+    def validate_tag_set_for_plate
+      return true if @plate.wells.count % tag_set.tags.count == 0
+
+      validation_errors << ['Tag set', 'tag count must be a factor of plate well count']
+      false
+    end
+
+    def try_get_sorted_wells(primary_grouping_direction)
+      if primary_grouping_direction == 'column'
+        @sorted_wells = @plate.wells_by_column_then_row
+        true
+      elsif primary_grouping_direction == 'row'
```

Comment:
I was wondering whether these grouping directions should be `horizontal` and `vertical`?  I'm just thinking `row` and `column` infers that the grouping is per-row or per-column and it's not.

Reply:
I'm happy to change these. I only think it makes sense to do so in the context of the factory though, i.e. the well still has a row and a column, and the plate still sorts by row then column etc.

Reply:
Yes absolutely.  The sorting of wells is row and column specific.  The creation of libraries is the right place for this.